packages/core/src/components/button/abstractButton.tsx

```diff
@@ -126,10 +126,10 @@ export abstract class AbstractButton<T> extends React.Component<React.HTMLProps<
         });
 
         return [
-            loading ? <Spinner className="pt-small pt-button-spinner" key="spinner" /> : undefined,
-            text != null ? <span key="text">{text}</span> : undefined,
-            children,
-            rightIconName != null ? <span className={iconClasses} key="icon" /> : undefined,
+            loading && <Spinner className="pt-small pt-button-spinner" key="spinner" />,
+            text != null && <span key="text">{text}</span>,
```

Comment:
the truthy value approach is probably more accurate cuz that's how react works internally (`false` renders nothing). i'll change to that.